Sources/NIO/EventLoop.swift

```diff
@@ -791,3 +763,77 @@ public enum EventLoopError: Error {
     /// Shutting down the `EventLoop` failed.
     case shutdownFailed
 }
+
+fileprivate struct TaskQueue {
+    private let thread = Thread.current
+    private var scheduledTasks = PriorityQueue<ScheduledTask>(ascending: true)
```

Comment:
these properties should have comments on what the protocol for accessing them is, for example for `pending*` it should state that that can only be accessed with `lock` taken